zkvm/tests/zkvm.rs

```diff
@@ -0,0 +1,81 @@
+use bulletproofs::BulletproofGens;
+use curve25519_dalek::scalar::Scalar;
+use hex;
+use merlin::Transcript;
+
+use zkvm::*;
+
+fn issue_contract(
+    qty: u64,
+    flv: Scalar,
+    issuance_pred: Predicate,
+    nonce_pred: Predicate,
+    recipient_pred: Predicate,
+) -> Vec<Instruction> {
+    vec![
+        Instruction::Push(
+            Commitment::from(CommitmentWitness {
+                value: qty.into(),
+                blinding: Scalar::from(1u64),
+            })
+            .into(),
+        ), // stack: qty
+        Instruction::Var, // stack: qty-var
+        Instruction::Push(Commitment::from(CommitmentWitness::unblinded(flv)).into()), // stack: qty-var, flv
+        Instruction::Var,                         // stack: qty-var, flv-var
+        Instruction::Push(issuance_pred.into()),  // stack: qty-var, flv-var, pred
```

Comment:
Fun idea: have an `ops::Builder` type so we can write `ops::build().push(issuance_pred).issue().push(nonce_pred).nonce().signtx()...` and make `push<T: Into<Data>>` so we don't even have to type `.into()` all the time.

Reply:
(Note for later)